src/jsonpatcherproxy.js

```diff
@@ -143,7 +144,9 @@ var JSONPatcherProxy = (function() {
         return Reflect.deleteProperty(target, key);
       }
     });
-    return proxy;
+    /* keeping track of all the proxies to be able to revoke them later */
+    this.proxifiedObjectsMap.push(proxy);
+    return proxy.proxy;
   };
   //grab tree's leaves one by one, encapsulate them into a proxy and return
   JSONPatcherProxy.prototype._proxifyObjectTreeRecursively = function(
```

Comment:
Could `_proxifyObjectTreeRecursively` automatically revoke proxy of when a child object is removed?